sdk/core/AzureCore/Source/Pipeline/Transport/URLSessionTransport.swift

```diff
@@ -106,6 +128,14 @@ public class URLSessionTransport: HTTPTransportStage {
                 logger: logger,
                 context: responseContext
             )
+            // Check if clientRequestId is canceled and back out if it is.
```

Comment:
We should probably revert this. At this point, the request has been sent already, it's too late to cancel - the operation is complete, so even if cancellation was requested, we should return back the completion to the caller. Not doing so could cause problems. For example, if the caller tried to create a resource, and then canceled the task after the request was already sent. The resource would be created by the service, but this logic here would make them believe the task had been cancelled successfully.

Reply:
Or in lieu of stating that "Request canceled" (since it wasn't) we change the error message to something like "Request cancellation failed," which I would argue is still an error. Requesting cancellation and then getting a success result would seem bizarre to me. Then again, the right answer is probably what do other SDKs do. @adrianhall do you have any thoughts here?

Reply:
After discussion with @JeffreyRichter, this is the correct behavior (albeit not necessarily the correct implementation). Cancellation is best effort and not guaranteed, but if they issue a cancellation request, they should get a cancellation "response". The key is to document this behavior clearly.